### DISCUSSION TITLE
How to use Handlebars.Net to templatize a html template if it has 2 grids/tables?

### DISCUSSION THREAD

**Discussion by kagexander** [Handlebars-Net/Handlebars.Net#549] [Q&A] [answered]
Lets say I have a html template like this

  {{#each countries}}
        Country Name : {{countryName}} ,Capital :{{countryCapital}}<br>
{{/each}}

{{#each names}}
	Name : {{firstName}} {{lastName}}<br>
{{/each}}

and in c# I will haev 2 datasets - countries & names. What should be the syntax to fill both the grids/loops ?

**Comment by rexm** [comment, marked as answer]
var result = template(new {
  countries = countryData,
  names = nameData
});

**Comment by kagexander** [comment]
I think what you have mentioned cannot be used in C#.
I have been using this piece of code in C#. And it does not produce the desired result.

var LoadedTemplate = Handlebars.Compile(html);
 var result = LoadedTemplate(countries , names);

//countries & Names are 2 lists with the data populated in it

**Comment by rexm** [comment]
What you are trying would never work. Handlebars expects you to pass a single model object, and it expects the names in the template to map to properties on the model. In your case even if it did support multiple models, they aren’t named, so how would it know the first one is “countries” and the second one is “names”?

As it is, the code example I shared above does indeed work in c# (I fixed a small typo).

**Comment by kagexander** [comment]
Thanks @rexm  for your explanation, can you help me understand how would the compiler understand wat object is 
new {
countries: countryData,
names: nameData
}

When i try to use , I keep these compile time issues.
![image](https://github.com/Handlebars-Net/Handlebars.Net/assets/22502297/8d154b52-ab5d-409f-9f45-5fbdf948806f)


If I seperately use different variable names , e.g. countryData,nameData - lists in C# , then wat will be countries & names? My understanding was that the same list name needs to be provided in the html template.

It could be that I am missing something. Thanks for your assistance though!!

**Comment by rexm** [comment]
Sorry it should be = not : I updated it

**Reply by kagexander** [reply]
Thanks it helped!!
